### PR TITLE
Fix persisting the Cobbler UID in the database via XML-RPC createSystemRecord (bsc#1207532)

### PR DESCRIPTION
**Pull request by meaksh**
## What does this PR change?

This is just a recreation of https://github.com/uyuni-project/uyuni/pull/6676 in a Uyuni branch.

## GUI diff

No difference.

- [x] **DONE**

## Documentation
- No documentation needed: only internal and user invisible changes

- [x] **DONE**

## Test coverage
- Unit tests were added

- [x] **DONE**

## Links

Tracks https://github.com/SUSE/spacewalk/issues/20333

- [ ] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

**Comment by mcalmer** [comment]
@jordimassaguerpla said yesterday the acceptance tests should work. But they still fail in the same way as in the last days. Do we need to rebase only or do we have a real problem?

**Comment by mcalmer** [comment]
@jordimassaguerpla I rebased and we still have all these errors. I do not think that the code here cause them. Anything known currently?

**Review by mcalmer — APPROVED** [review]
LGTM

File: java/code/src/org/cobbler/CobblerObject.java [hunk 1]
```diff
@@ -171,7 +171,7 @@ protected static List<Map<String, Object>> lookupDataMapsByCriteria(
         Map<String, String> criteria = new HashMap<>();
         criteria.put(critera, value);
         return (List<Map<String, Object>>)
-                client.invokeTokenMethod(findMethod, criteria);
+                client.invokeTokenMethod(findMethod, criteria, false);
```
Comment (mcalmer) [inline java/code/src/org/cobbler/CobblerObject.java:174]: This seems to be wrong. Paramater needs to be "true": See also https://github.com/SUSE/spacewalk/pull/22954

**Review by mcalmer — CHANGES_REQUESTED** [review]

File: java/code/src/org/cobbler/CobblerObject.java (hunk 1 above)
Comment (meaksh) [reply java/code/src/org/cobbler/CobblerObject.java]: @mcalmer I see you have added few more commits to this PR after you did this review.

Is this PR ready to be merged now?

Thanks in advance!

**Review by mcalmer — APPROVED** [review]
LGTM

**Comment by meaksh** [comment]
The current failure we see in `acceptances-tests` is not really caused by this PR. It is actually a timing issue in the testsuite:

```
Text '28 minutes ago' not found (ScriptError)
```
While screenshot shows `27 minutes ago`.

So I'm merging this PR now.

**Pull request merged**